homeassistant/components/sensor/gitlab_ci.py

```diff
@@ -0,0 +1,182 @@
+"""Module for retrieving latest GitLab CI job information."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_SCAN_INTERVAL, CONF_TOKEN, CONF_URL)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+CONF_GITLAB_ID = 'gitlab_id'
+CONF_ATTRIBUTION = "Information provided by https://gitlab.com/"
+
+ICON_HAPPY = 'mdi:emoticon-happy'
+ICON_SAD = 'mdi:emoticon-happy'
+ICON_OTHER = 'mdi:git'
+
+ATTR_BUILD_ID = 'build id'
+ATTR_BUILD_STATUS = 'build_status'
+ATTR_BUILD_STARTED = 'build_started'
+ATTR_BUILD_FINISHED = 'build_finished'
+ATTR_BUILD_DURATION = 'build_duration'
+ATTR_BUILD_COMMIT_ID = 'commit id'
+ATTR_BUILD_COMMIT_DATE = 'commit date'
+ATTR_BUILD_BRANCH = 'master'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_GITLAB_ID): cv.string,
+    vol.Optional(CONF_URL, default='https://gitlab.com'): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL,
+                 default=timedelta(seconds=300)): cv.time_period,
+})
+
+REQUIREMENTS = ['python-gitlab==1.6.0']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Sensor platform setup."""
+    _priv_token = config.get(CONF_TOKEN)
+    _gitlab_id = config.get(CONF_GITLAB_ID)
+    _interval = config.get(CONF_SCAN_INTERVAL)
+    _url = config.get(CONF_URL)
+
+    _gitlab_data = GitLabData(
+        priv_token=_priv_token,
+        gitlab_id=_gitlab_id,
+        interval=_interval,
+        url=_url
+    )
+
+    add_devices([GitLabSensor(_gitlab_id, _priv_token, _gitlab_data)], True)
+
+
+class GitLabSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, gitlab_id, priv_token, gitlab_data):
+        """Initialize the sensor."""
+        self._state = None
+        self._available = False
+        self._status = None
+        self._started_at = None
+        self._finished_at = None
+        self._duration = None
+        self._commit_id = None
+        self._commit_date = None
+        self._build_id = None
+        self._branch = None
+        self._gitlab_data = gitlab_data
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return 'GitLab CI Status'
```

Comment:
It would be good to allow the user to name the sensor if the user configures multiple platform instances.